test/org/sagebionetworks/bridge/services/StudyServiceTest.java

```diff
@@ -61,31 +82,107 @@
     @Resource
     SubpopulationDao subpopDao;
 
+    @Resource
+    SynapseClient synapseClient;
+
     @Autowired
     CacheProvider cache;
     
     private CacheProvider mockCache;
     
     private Study study;
+
+    private Project project;
+    private Team team;
     
     @Before
     public void before() {
         mockCache = mock(CacheProvider.class);
         studyService.setCacheProvider(mockCache);
+
+        studyService.setSynapseClient(synapseClient);
     }
     
     @After
-    public void after() {
+    public void after() throws SynapseException {
         if (study != null) {
             studyService.deleteStudy(study.getIdentifier());
         }
+        if (project != null) {
+            synapseClient.deleteEntityById(project.getId());
+        }
+        if (team != null) {
+            synapseClient.deleteTeam(team.getId());
+        }
     }
 
     @After
     public void resetCache() {
         studyService.setCacheProvider(cache);
```

Comment:
![image](https://cloud.githubusercontent.com/assets/22622093/20775750/5a8fb118-b711-11e6-889d-e299ecb7820d.png)
I think this line already did that